source/Calamari.AzureAppService/Behaviors/AzureAppServiceZipDeployBehaviour.cs

```diff
@@ -126,33 +126,53 @@ public async Task Execute(RunningDeployment context)
              * https://github.com/OctopusDeploy/Calamari/tree/master/source/Calamari.Common/Features/Behaviours
              */
             var uploadPath = string.Empty;
-            if (substitutionFeatures.Any(featureName => context.Variables.IsFeatureEnabled(featureName)))
-                uploadPath = (await packageProvider.PackageArchive(context.StagingDirectory, context.CurrentDirectory)).FullName;
-            else
-                uploadPath = (await packageProvider.ConvertToAzureSupportedFile(packageFileInfo)).FullName;
-
-            if (uploadPath == null)
-                throw new Exception("Package File Path must be specified");
-
-            // need to ensure slot is created as slot creds may be used
-            if (targetSite.HasSlot && slotCreateTask != null)
-                await slotCreateTask;
-
-            Log.Verbose($"Retrieving publishing profile for App Service to determine correct deployment endpoint.");
-            using var publishingProfileXmlStream = await armClient.GetPublishingProfileXmlWithSecrets(targetSite);
-            var publishingProfile = await PublishingProfile.ParseXml(publishingProfileXmlStream);
-
-            Log.Verbose($"Using deployment endpoint '{publishingProfile.PublishUrl}' from publishing profile.");
-
-            Log.Info($"Uploading package to {targetSite.SiteAndSlot}");
-
-            if (packageProvider.SupportsAsynchronousDeployment && FeatureToggle.AsynchronousAzureZipDeployFeatureToggle.IsEnabled(context.Variables))
+            bool uploadFileNeedsCleaning = false;
```

Comment:
Will this be required in the LegacyAzureAppServiceBehaviour as well?

Reply:
good question, let me check

Reply:
yes, missed that, thanks for the catch